.github/workflows/docker_build_and_push.yml

```diff
@@ -0,0 +1,75 @@
+name: CI/CD Pipeline for PersistenceCore
+
+on:
+  push:
+    branches:
+      - 'main'
```

Comment:
I think we shouldn't push images on docker hub that are not associated with a release.